br/pkg/lightning/restore/tidb.go

```diff
@@ -373,8 +374,12 @@ func ObtainNewCollationEnabled(ctx context.Context, g glue.SQLExecutor) (bool, e
 // NOTE: since tidb can make sure the auto id is always be rebase even if the `incr` value is smaller
 // the the auto incremanet base in tidb side, we needn't fetch currently auto increment value here.
 // See: https://github.com/pingcap/tidb/blob/64698ef9a3358bfd0fdc323996bb7928a56cadca/ddl/ddl_api.go#L2528-L2533
-func AlterAutoIncrement(ctx context.Context, g glue.SQLExecutor, tableName string, incr int64) error {
-	logger := log.With(zap.String("table", tableName), zap.Int64("auto_increment", incr))
+func AlterAutoIncrement(ctx context.Context, g glue.SQLExecutor, tableName string, incr uint64) error {
+	logger := log.With(zap.String("table", tableName), zap.Uint64("auto_increment", incr))
+	if incr > math.MaxInt64 {
+		logger.Warn("auto_increment out of the maximum value TiDB supports", zap.Uint64("auto_increment", incr))
+		return nil
+	}
```

Comment:
Do we need to set `auto_increment` to math.MaxInt64`?

Reply:
MaxInt64 is a legal value in TiDB (i.e. you can insert (9223372036854775807, ....) to a table) whereas MaxInt64+1 is illegal, which will trigger the error, `ERROR 1467 (HY000): Failed to read auto-increment value from storage engine`.

Reply:
Does `incr > math.MaxInt64` means lightning has inserted rows whose id is greater than `math.MaxInt64`? Will this cause duplicate entry or data corruption?

Reply:
Mostly yes. But in terms of the case shown in the linked issue, `9223372036854775807` is a valid input, but after inserting it, Lightning would try `alter table xxx auto_increment=9223372036854775807+1`, get syntax error from TiDB, and fail unexpectedly.

So I'm not trying to legalize auto-incr value that exceeds math.MaxInt64 but try to skip this syntax error and let Lightning fail by other clearer issues (such as data corruption, duplicate entry, etc.). As for the case in this issue, because Lightning doesn't try to write entries that are larger than 9223372036854775807, it will succeed.

Failing by duplicate entries might be more intuitive in terms of the effectiveness of reporting the error. Syntax error is helpless...

Reply:
Could confirm whether data corruption or any other error can be reported when using local backend and auto_inc is overflow?

Reply:
Yes. If the `auto_incr` column overflows, the checksum will be mismatched.

Reply:
"checksum mismatch" is hard for user to know what happened. How about checking overflow during encoding and give a clear error to user?

Reply:
> "checksum mismatch" is hard for user to know what happened. How about checking overflow during encoding and give a clear error to user?

Yes. Actually I'm looking at it #28776. This PR only solves the unexpected syntax error.